### PR TITLE
Fixes:#2 Add DeepFace-based emotion detection with YouTube music recommendation 

### PR DESCRIPTION
**Pull request by vanixyz**
📂 What I’ve Done:
✅ Created a new version of the Streamlit app as requested:
• Copied from: /code/ui_interfaces/app_local_streamlit.py
• New file: /code/new_models/app_local_streamlit_new_model_1.py

✅ Integrated DeepFace for facial emotion recognition.
✅ Used detected emotion to stream relevant music via YouTube.
✅ Improved UI with dynamic themes and emotion-based color feedback..

📌 About the Model:
I am using DeepFace’s built-in pre-trained model (Mini-Xception on FER2013) via DeepFace.analyze() with actions=["emotion"].

❗ No .hdf5 file was added manually — DeepFace manages this internally.

➕ If needed, I can place a placeholder README in /code/model/ to clarify.

Comparison: DeepFace vs. .hdf Model File Approach
✅ Ease of Use
.hdf: Requires manual loading using load_model(), manual image preprocessing, and inference.

DeepFace: One-liner functions like DeepFace.verify() and DeepFace.analyze() handle everything internally.

✅ Model Format & Flexibility
.hdf: Typically tied to a single custom model you maintain.

DeepFace: Offers multiple built-in models — VGG-Face, Facenet, ArcFace, SFace, Dlib, etc.

✅ Integration & Maintenance
.hdf: You need to write and manage the entire inference pipeline.

DeepFace: Abstracted APIs, prebuilt pipelines, and maintained by the open-source community.

✅ Performance
.hdf: Performance depends on how optimized your custom model is.

DeepFace: Optimized and tested across backends (TensorFlow, PyTorch), with consistently good speed and accuracy.

✅ Preprocessing & Utilities
.hdf: You must write preprocessing steps (resizing, normalization, alignment, etc.).

DeepFace: Automatically handles preprocessing, detection, and alignment.

✅ Deployment
.hdf: May require larger files and more setup.

DeepFace: Lightweight, self-contained, and easier to deploy for many use cases.

🧪 How to Test:
# Activate environment
venv\Scripts\activate

# Run the app
streamlit run code/new_models/app_local_streamlit_new_model_1.py
🔧 Dependencies:
Make sure these packages are installed:
pip install deepface opencv-python streamlit requests

Here’s a quick screenshot of the updated app in action with DeepFace integration:

🎯 Detected Emotion: happy
🎵 Now Playing: YouTube music based on emotion

Screenshot (1204)

📊 Benchmark Comparison Summary
Model	Inference Time	Detected Emotion	Accuracy Observation
DeepFace	0.587s	Happy	✅ Correct Detection
HDF5 Model	0.716s	Fear	❌ Incorrect (User was smiling) 

📝 The HDF5 model misclassified a clearly happy emotion as fear, whereas DeepFace correctly detected the expression.
🔍 Despite DeepFace being slightly faster in this test, its key strength is improved accuracy and real-time capability backed by more robust pre-trained weights and face analysis logic.

Screenshot (1206)
Screenshot (1207)

Thanks for the opportunity to contribute! 😊

### CODE REVIEW

**Review by SGCODEX — CHANGES_REQUESTED** [review]
Heyy, @vanixyz!! Great work, The PR is well explained and code looks good. Please attach some screenshots to show how this is working and also if you can so please compare and tell here how is this library better or more efficient than the hdf model file way. Also can maybe test both against some general benchmarks and share those results too.

**Comment by vanixyz** [comment]
Heyy @SGCODEX ,I have made the neccesary changes in the PR kindly check and let me know if I need to commit the benchmarking code too for both the files 😊 . 
Thankyou.

**Comment by SGCODEX** [comment]
Heyy @vanixyz , That's actually so great, the benchmark makes the difference so clear and much more accurate. Please commit the benchmark code too, would love to read and incorporate that too.

**Review by SGCODEX — CHANGES_REQUESTED** [review]
Please do the above. Thank you.

**Comment by vanixyz** [comment]
Heyy @SGCODEX  I have commited the benchmark code too . 
It's been a great experience contributing .
Thankyou 😊

**Review by SGCODEX — CHANGES_REQUESTED** [review]
Heyy @vanixyz, That looks great, well done really!! Just a small change.

Please copy `/code/ui_interfaces/app_local_streamlit.py` file in `/code/new_models/` directory and make the benchmark changes there and rename that file as `app_local_streamlit_with_benchmark.py`. Please revert back the changes made to `/code/ui_interfaces/app_local_streamlit.py`. The new_model_1 file looks great, no changes required in that right now. Thank you and really great work!!👏👏

**Comment by vanixyz** [comment]
Thankyou @SGCODEX  I have done the required changes  please review them

**Review by SGCODEX — CHANGES_REQUESTED** [review]
Heyy @vanixyz, can you try syncing fork once and there are still changes in the original `/code/ui_interfaces/app_local_streamlit.py` file, please revert it back to the original code. It will cause issues during merging. Thank you. Rest looks good.

And sorry for so many changes, also please rename `code/new_models/app_local_streamlit_new_model_1.py` as `code/new_models/app_local_streamlit_deepface.py`

Thank you so much

image

Ignore: #2

**Comment by vanixyz** [comment]
Hiii @SGCODEX  I am sorry i missed that could you 
please review it again .

**Review by SGCODEX — CHANGES_REQUESTED** [review]
image

Heyy @vanixyz , The code seems unchanged from last review. app_local_streamlit.py still shows in modified file and rename is still not there. Please can you check again, Thank you. (And also no sorry, its all good :) )

**Comment by vanixyz** [comment]
Hey @SGCODEX  thankyou for being so cooperative , I hope it works well now 😊

**Review by SGCODEX — CHANGES_REQUESTED** [review]
Heyy @vanixyz , Really commendable work!! It's superfast and way more accurate. I tried running it on my system and found some version errors, but here are the fixes for those, please comment these versions at the top of the file for other contributors to know as well. 
`# protobuf==3.20.3, tensorflow==2.10`
Also can you add bounding box and emotion detected on the live cam feed too?
Rest looks amazing. Thanks.

Ignore #2

**Comment by vanixyz** [comment]
hey @SGCODEX I have 
 Add bounding box and emotion label overlay on webcam feed
Enhanced detect_emotion() to draw bounding boxes and emotion labels above detected faces

Added version requirement notes at top (protobuf==3.20.3, tensorflow==2.10)

Verified live detection works with visual feedback and emotion-based music selection
Screenshot (1270)


Ready to merge into SGCODEX:contribution. Let me know if anything else is needed. 👍

**Comment by vanixyz** [comment]
hey @SGCODEX kindly review it once .

**Comment by vanixyz** [comment]
hey  @SGCODEX its been a while you haven't approved it kindly do it  please

GSSOC'25 contributor
